lib/database_validations/lib/adapters/base_adapter.rb

```diff
@@ -9,18 +9,25 @@ def initialize(model)
       end
 
       # @param [String] index_name
-      def find_index_by_name(index_name)
-        indexes.find { |index| index.name == index_name }
+      def find_unique_index_by_name(index_name)
+        unique_indexes.find { |index| index.name == index_name }
       end
 
       # @param [Array<String>] columns
       # @param [String] where
-      def find_index(columns, where)
-        indexes.find { |index| Array.wrap(index.columns).map(&:to_s).sort == columns && index.where == where }
+      def find_unique_index(columns, where)
+        unique_indexes.find { |index| Array.wrap(index.columns).map(&:to_s).sort == columns && index.where == where }
       end
 
-      def indexes
-        model.connection.indexes(model.table_name).select(&:unique)
+      def unique_indexes
```

Comment:
Do we want to cache it on our own to support old versions of Active Record?

Reply:
Not really, they should upgrade. Also, this is not big improvement as the query is performed only after constraint is violated.